src/modules/account/RadixAccount.ts

```diff
@@ -183,15 +220,97 @@ export default class RadixAccount {
     }
 
     private _onAtomReceived = async (atomObservation: RadixAtomObservation) => {
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+
+
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() + 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
 
         atomObservation.processedData = {}
         for (const system of this.accountSystems.values()) {
             await system.processAtomUpdate(atomObservation)
         }
 
-        this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        if (!this.processingAtomCounter.closed && !this.processingAtomCounter.isStopped) {
+            this.processingAtomCounter.next(this.processingAtomCounter.getValue() - 1)
+        } else {
+            logger.error(`☢️ processingAtomCounter closed or stopped`)
+        }
+
     }
 
-    
+    public async requestRadsForDevelopmentFromFaucetService(): Promise<RadixTransaction> {
+        return this.requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate()
+            .then((txUnique) => {
+                return this.transferSystem.getAllTransactions()
+                    .delay(2_000)
+                    .take(1)
+                    .timeout(60_000)
+                    .pipe(catchError((error, obs) => {
+                        throw new Error(`🚰 Failed to update balance after having received test XRD tokens from
+                              faucet in tx with unique string
+                               ${txUnique}, error: ${error}
+                               `)
+                    }))
+                    .pipe(map((tu: RadixTransactionUpdate) => tu.transaction))
+                    .toPromise()
+            },
+            ).catch(errorFromFaucet => {
+                const errorMsg = `Faucet server is down? ${errorFromFaucet}`
+                logger.error(errorMsg)
+                throw new Error(errorMsg)
+            })
+    }
+
+    public async requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate(
+        maxNumberOfRetries: number = 10,
+        killNodeServerOfFailure: boolean = true,
+    ): Promise<string> {
+
+        for (let attempt = 0; attempt < maxNumberOfRetries; attempt++) {
+            const idOfTxFromFaucet = await RadixAccount.getTokensFromFaucetURL(this.address)
+            if (idOfTxFromFaucet !== undefined) {
```

Comment:
can be shortened to `if (idOfTxFromFaucet)`

Reply:
👍